Controller

zipnikitin opened the pull request:
1. Создайте ресурс Вопросов, вложенный в ресурс Тестов - создано в routes.rb

2. Следуя принципам  REST, cоздайте контроллер для ресурса вопросов и реализуйте методы:
Просмотра списка вопросов теста - index
Просмотра конкретного вопроса теста - show
Создания вопросa - create
Удаления вопроса - destroy
Используйте метод render внутри действия контроллера чтобы вернуть ответ клиенту как простой текст (plain text) или HTML - сделал
Используйте обратный вызов при поиске теста в контроллере ресурса вопросов - before_action :test_search

3. Напишите обработку исключения для случая когда вопрос не был найден - question_not_found

zipnikitin committed:
Controller

1. Создайте ресурс Вопросов, вложенный в ресурс Тестов - создано в routes.rb

2. Следуя принципам  REST, cоздайте контроллер для ресурса вопросов и реализуйте методы:
Просмотра списка вопросов теста - index
Просмотра конкретного вопроса теста - show
Создания вопросa - create
Удаления вопроса - destroy
Используйте метод render внутри действия контроллера чтобы вернуть ответ клиенту как простой текст (plain text) или HTML - сделал
Используйте обратный вызов при поиске теста в контроллере ресурса вопросов - before_action :test_search

3. Напишите обработку исключения для случая когда вопрос не был найден - question_not_found

VidgarVii commented on testGuru/app/controllers/questions_controller.rb:
```diff
@@ -0,0 +1,41 @@
+
+    if question.save
+      render plain: @question.body.inspect
+    end
```
1 разберись отличе м/у redirect_to и render. В rusrails есть инофрмация
2 если ресурс не сохранился. Юзеру надо обновить форму. И показать ошибки валидации(показ ошибок сделаем на след ДЗ)

VidgarVii commented on testGuru/app/controllers/questions_controller.rb:
```diff
@@ -0,0 +1,41 @@
+  def destroy
+    @question.delete
+
+    render plain: '<%= @test.questions.inspect %> удален'
```
тут тоже ошибся с рендером. Лучше юзера отправить на какой-нить список вопросов

VidgarVii commented on testGuru/config/routes.rb:
```diff
@@ -1,3 +1,8 @@
+    resources :questions, shallow: true
+  end
+  
+  get 'questions/:id/delete', to: 'questions#destroy'
```
этот рот лишний.
Для всех CRUD операций все пути у тебя уже есть.
Посмотри rails routes в консоле и исправь контроллер

VidgarVii commented on testGuru/app/controllers/questions_controller.rb:
```diff
@@ -0,0 +1,41 @@
+  end
+
+  def destroy
+    @question.delete
```
от куда тут появится вопрос?

VidgarVii commented on testGuru/app/controllers/questions_controller.rb:
```diff
@@ -0,0 +1,41 @@
+  end
+
+  def show
+    render plain: @question.body.inspect
```
от куда тут появится вопрос?

zipnikitin committed:
Controller 2

Понял, куда и как применить redirect_to, отправляет на список тестов, где это нужно
Поправил routes, больше ничего лишнего
Вспомнил, что нужно объявить переменную экземпляра класса Question, создал метод и фильтр

VidgarVii commented on testGuru/app/controllers/questions_controller.rb:
```diff
@@ -0,0 +1,48 @@
+    question = @test.questions.new(question_params)
+
+    if question.save
+      redirect_to action: :index
```
используй объет redirect_to question
а лучше @question

zipnikitin committed:
Delete .DS_Store

zipnikitin committed:
Delete .DS_Store
